web/packages/teleport/src/UnifiedResources/UnifiedResources.tsx

```diff
@@ -62,6 +64,31 @@ export function UnifiedResources() {
   );
 }
 
+const getAvailableKindsWithAccess = (flags: FeatureFlags): FilterKind[] => {
+  return [
+    {
+      kind: 'node',
+      disabled: !flags.nodes,
+    },
+    {
+      kind: 'app',
+      disabled: !flags.applications,
+    },
+    {
+      kind: 'db',
+      disabled: !flags.databases,
+    },
+    {
+      kind: 'kube_cluster',
+      disabled: !flags.kubernetes,
+    },
+    {
+      kind: 'windows_desktop',
+      disabled: !flags.desktops,
+    },
```

Comment:
hrm, so if we do `action` checks, historically (look at `getXXX` actions), i think all these resources check for both a `list` and a `read` action to be able to query (except for nodes, i think nodes is the only exception to this rule for [unknown reasons](https://github.com/gravitational/teleport/blob/e9e8760f72ecf7f61f6bf033155b00e606322a62/lib/auth/auth_with_roles.go#L1856))

 i think all these flags only returns you  a `list`. I know this existed before, and highly unlikely that if you don't have list, why missing read and vice versa, but perhaps we should be thorough anyways (to at least match the backend) and also check for read?

Reply:
I gave this a shot (its a bit ugly). I might have misunderstood the ask, but is [this commit](https://github.com/gravitational/teleport/pull/34205/commits/a83299b5e40a1360ad0f8d5d937e47a2643cf0a9) doing what you are asking?